File: Raven.Client.Java/lib/src/main/java/raven/abstractions/data/JsonDocumentMetadata.java
package raven.abstractions.data;

import java.util.Date;

import raven.abstractions.json.linq.RavenJObject;

public class JsonDocumentMetadata implements IJsonDocumentMetadata {

  private RavenJObject metadata;
  private String key;
  private Boolean nonAuthoritativeInformation;
  private Etag etag;
  private Date lastModified;

  /**
   * @return the etag
   */
  public Etag getEtag() {
    return etag;
  }

  /**
   * @return the key
   */
  public String getKey() {
    return key;
  }

  /**
   * @return the lastModified
   */
  public Date getLastModified() {
    return lastModified;
  }

  /**
   * @return the metadata
   */
  public RavenJObject getMetadata() {
    return metadata;
  }

  /**
   * @return the monAuthoritativeInformation
   */
  public Boolean getNonAuthoritativeInformation() {
    return nonAuthoritativeInformation;
  }

  /**
   * @param etag the etag to set
   */
  public void setEtag(Etag etag) {
    this.etag = etag;
  }

  /**
   * @param key the key to set
   */
  public void setKey(String key) {
    this.key = key;
  }

  /**
   * @param lastModified the lastModified to set
   */
  public void setLastModified(Date lastModified) {
    this.lastModified = lastModified;
  }

  /**
   * @param metadata the metadata to set
   */
  public void setMetadata(RavenJObject metadata) {
    this.metadata = metadata;
  }

  /**
   * @param monAuthoritativeInformation the nonAuthoritativeInformation to set
   */
  public void setNonAuthoritativeInformation(Boolean nonAuthoritativeInformation) {
    this.nonAuthoritativeInformation = nonAuthoritativeInformation;
  }

}

File: src/Raven.Client/Data/JsonDocumentMetadata.cs
using System;
using Raven.Json.Linq;

namespace Raven.Client.Data
{
    /// <summary>
    /// A document representation:
    /// * Etag
    /// * Metadata
    /// </summary>
    public class JsonDocumentMetadata : IJsonDocumentMetadata
    {
        /// <summary>
        /// Metadata for the document
        /// </summary>
        public RavenJObject Metadata { get; set; }

        /// <summary>
        /// Key for the document
        /// </summary>
        public string Key { get; set; }

        /// <summary>
        /// Current document etag.
        /// </summary>
        public long? Etag { get; set; }

        /// <summary>
        /// Last modified date for the document
        /// </summary>
        public DateTime? LastModified { get; set; }
    }
}
